### ISSUE TITLE
Core/Creature: Creatures follows player in each other instead of using random angle and distance

### ISSUE THREAD

**Issue by offl** [TrinityCore/TrinityCore#25063]
**Description:**

If multiple creatures follows player and they are not guardians, they follows player by using the same angle and distance, in other words they follows inside each other and not like in https://www.youtube.com/watch?v=LLWlj38eRZ4

**Expected behaviour:**

If multiple creatures follows player they should use random dist and angle and not follow player in each other
They should not act like guardians(it's implemented to guardians but they uses different system, see https://youtu.be/jWlqGVyb_mQ?t=175)
Maybe we should add an param to SAI to allow follow with random dist and angle or it must be done in core for all following creatures, for example for mini-pets

**Steps to reproduce the problem:**

1. apply https://github.com/TrinityCore/TrinityCore/issues/25062
2. .q a 11560
3. .go c id 25201
4. click on the cage and move around
5. notice murlocs follows in each other

**Branch:** 3.3.5
**TC rev. hash/commit:** 8675fd756ff9

**Comment by offl** [comment]
Or try with https://github.com/TrinityCore/TrinityCore/issues/25064

**Comment by natnat201193** [comment]
This is due to creature spreading being missing on TC https://github.com/TrinityCore/TrinityCore/issues/18512 this issue is about combat, but its the same issue

**Comment by illfated** [comment]
Fair point. I wonder if this belongs only in the Creature AI section or both in the combat code & creature AI. Anyway, I am pretty sure it needs to be done in core, to enable the NPCs to check their distance to other creatures in the same aggro/threat table or if they are reacting to the same player creature(s). Would be fun to see a meta-scripting for the behaviour, though.
